### PR TITLE
add '(or subset)' after config

### PR DESCRIPTION
**Pull request by severo**
cc https://github.com/huggingface/datasets/pull/7057#pullrequestreview-2191208969

> cool, let's aim to have "subsets" everywhere, and "configure the subsets" when it comes to the YAML (that uses the fields "configs" / "config_name" ) :)

### CODE REVIEW

**Comment by lhoestq** [comment]
Maybe we should only use subset no ? in most cases it's ok, and in the remaining cases when the docs refers to the "configs" in YAML we can just formulate it as "configure your subsets" / "define one configuration per subset" ?

**Comment by severo** [comment]
OK. I'll do this

File: datasetcard.md [hunk 1]
```diff
@@ -34,13 +34,13 @@ task_ids:
 - {subtask_1}  # Example: multi-class-image-classification
 paperswithcode_id: {paperswithcode_id}  # Dataset id on PapersWithCode (from the URL). Example for SQuAD: squad
 configs:  # Optional. This can be used to pass additional parameters to the dataset loader, such as `data_files`, `data_dir`, and any builder-specific parameters  
-- config_name: {config_name_0}  # Example: default
+- config_name: {config_name_0}  # Name of the configuration (or subset). Example: default
```
Comment (julien-c) [inline datasetcard.md:37]:
```suggestion
- config_name: {config_name_0}  # Name of the dataset subset, if applicable. Example: default
```

you mean like this @lhoestq?

File: datasetcard.md [hunk 2]
```diff
@@ -69,15 +69,15 @@ dataset_info:
...
+  config_name: {config_name}  # Name of the configuration (or subset). Example for glue: sst2
   splits:
     - name: {split_name_0}                  # Example: train
       num_bytes: {split_num_bytes_0}        # Example for SQuAD: 79317110
       num_examples: {split_num_examples_0}  # Example for SQuAD: 87599
   download_size: {dataset_download_size}   # Example for SQuAD: 35142551
   dataset_size: {dataset_size}             # Example for SQuAD: 89789763
 
-# It can also be a list of multiple configurations:
+# It can also be a list of multiple configurations (or subsets):
```
Comment (julien-c) [inline datasetcard.md:80]: here i think it's fine this way

**Comment by severo** [comment]
I just remove 'configuration' nearly everywhere, and I think it's OK, even clearer. wdyt?

File: docs/hub/datasets-image.md [hunk 3]
```diff
@@ -153,7 +153,7 @@ You can disable this automatic addition of the `label` column in the [YAML confi
 
 ```yaml
 configs:
-  - config_name: default
+  - config_name: default # name of the dataset subset
```
Comment (lhoestq) [inline docs/hub/datasets-image.md:156]:
```suggestion
  - config_name: default  # Name of the dataset subset, if applicable.
```

**Review by lhoestq — APPROVED** [review]
lgtm !

File: datasetcard.md (hunk 2 above)
Comment (julien-c) [reply datasetcard.md:80]: or this:
> subsets (also called "configurations")

Comment (severo) [reply datasetcard.md:80]: https://github.com/huggingface/hub-docs/pull/1348

**Pull request merged**